### PR TITLE
[CI] Add diff print to CMake style check

### PR DESCRIPTION
**Pull request by KFilipek**
### Description

Example:
image

Source:
https://github.com/KFilipek/unified-memory-framework/actions/runs/9681607789/job/26712591718

### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

**Review by lplewa — APPROVED** [review]

File: .github/workflows/pr_push.yml [hunk 1]
```diff
@@ -141,7 +141,14 @@ jobs:
       run: cmake --build build --target clang-format-check
 
     - name: Check CMake formatting
-      run: cmake --build build --target cmake-format-check
+      run: |
+        cmake --build build --target cmake-format-apply
+        if ! git diff | wc -l | grep -q "^0$"; then
```
Comment (lukaszstolarczuk) [inline .github/workflows/pr_push.yml:146]: `git diff --exit-code`

File: .github/workflows/pr_push.yml [hunk 2]
```diff
@@ -141,7 +141,14 @@ jobs:
...
+          echo "Errors found:"
+          git diff
+          exit 1  # Indicate failure
+        fi
+        cmake --build build --target cmake-format-check
```
Comment (PatKamin) [inline .github/workflows/pr_push.yml:151]: Is this line still needed? You exit on failure already.

File: .github/workflows/pr_push.yml (hunk 1 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml]: Done

File: .github/workflows/pr_push.yml (hunk 2 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml]: Done

File: .github/workflows/pr_push.yml [hunk 3]
```diff
@@ -141,7 +141,10 @@ jobs:
       run: cmake --build build --target clang-format-check
 
     - name: Check CMake formatting
-      run: cmake --build build --target cmake-format-check
+      run: |
+        cmake --build build --target cmake-format-apply
+        echo "Errors found:"
+        git diff --exit-code
```
Comment (PatKamin) [inline .github/workflows/pr_push.yml:147]: `|| echo "no errors"` to leave no room for an empty line interpretation after the above echo.

File: .github/workflows/pr_push.yml [hunk 4]
```diff
@@ -141,7 +141,10 @@ jobs:
       run: cmake --build build --target clang-format-check
 
     - name: Check CMake formatting
-      run: cmake --build build --target cmake-format-check
+      run: |
+        cmake --build build --target cmake-format-apply
+        echo "Errors found:"
```
Comment (lukaszstolarczuk) [inline .github/workflows/pr_push.yml:146]: heh, is this `echo` really needed? now we print this, even though there were no errors

if you want this message printed it should be under `if` or e.g. like this:
`git diff --exit-code && echo "Errors found! || echo "positive message"`

Comment (PatKamin) [reply .github/workflows/pr_push.yml]: I agree that this message is not needed at all.

Comment (KFilipek) [reply .github/workflows/pr_push.yml]: Fail + diff output is sufficient. Echo removed.

**Review by PatKamin — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**